ray-curator/ray_curator/stages/text/io/reader/parquet.py

```diff
@@ -0,0 +1,129 @@
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Parquet reader composite stage."""
+
+from dataclasses import dataclass, field
+from typing import Any, Literal
+
+import pandas as pd
+
+from ray_curator.stages.base import CompositeStage
+from ray_curator.stages.file_partitioning import FilePartitioningStage
+from ray_curator.tasks import DocumentBatch, _EmptyTask
+
+from .base import BaseReader
+
+
+@dataclass
+class ParquetReaderStage(BaseReader):
+    """
+    Stage that processes a group of Parquet files into a DocumentBatch.
+    This stage accepts FileGroupTasks created by FilePartitioningStage
+    and reads the actual file contents into DocumentBatches.
+
+    Args:
+        fields (list[str], optional): If specified, only read these columns. Defaults to None.
+        read_kwargs (dict[str, Any], optional): Keyword arguments for the underlying reader. Defaults to {}.
+    """
+
+    _name: str = "parquet_reader"
+
+    def read_data(
+        self,
+        paths: list[str],
+        read_kwargs: dict[str, Any] | None = None,
+        fields: list[str] | None = None,
+    ) -> pd.DataFrame:
+        """Read Parquet files using Pandas. Raises an exception if reading fails."""
+
+        # Normalize read_kwargs to a dict to avoid TypeError when None
+        # Work on a copy to avoid mutating caller's dict
+        read_kwargs = {} if read_kwargs is None else dict(read_kwargs)
+
+        update_kwargs = {}
+        if fields is not None:
+            update_kwargs["columns"] = fields
+        if "engine" not in read_kwargs:
+            update_kwargs["engine"] = "pyarrow"
+        if "dtype_backend" not in read_kwargs:
+            update_kwargs["dtype_backend"] = "pyarrow"
+        read_kwargs.update(update_kwargs)
+        return pd.read_parquet(paths, **read_kwargs)
+
+
+@dataclass
+class ParquetReader(CompositeStage[_EmptyTask, DocumentBatch]):
+    """Composite stage for reading Parquet files.
+
+    This high-level stage decomposes into:
+    1. FilePartitioningStage - partitions files into groups
+    2. ParquetReaderStage - reads file groups into DocumentBatches
+    """
+
+    file_paths: str | list[str]
+    files_per_partition: int | None = None
+    blocksize: int | str | None = None
+    fields: list[str] | None = None  # If specified, only read these columns
+    read_kwargs: dict[str, Any] | None = None
+    file_extensions: list[str] = field(default_factory=lambda: [".parquet"])
```

Comment:
Does ".part" count here too?

Reply:
I think if its parquet, lets ask users to add `.part` ?  What do you think ?